### PR TITLE
TPD-940 Add CreateTinyUrl to PythonModel

### PR DESCRIPTION
**Pull request by davidcarroll**
Ability to create short urls inside a python script.
touchpointsoftware.atlassian.net/browse/TPD-940

### CODE REVIEW

File: UnitTests/CmsDataTests/API/PythonModel/PythonModelTests.cs [hunk 1]
```diff
@@ -120,6 +123,30 @@ public void Python_Result_Should_Have_All_SQL_Results()
...
+        public void CreateTinyUrlTest()
+        {
+            string id = null;
+            using (AppConfig.Change(GetWebConfig()))
+            {
+                // For this test to succeed,
+                // it will require adding a setting in appSettings of secrets.config (so as to not be included in source repo) under CMSWeb 
+                // The setting should be of the form: 
+                // <add key="tpsdbcotoken" value="secret token here" />
+                id = PythonModel.CreateTinyUrl("https://status.touchpointsoftware.com");
```
Comment (brondavies) [inline UnitTests/CmsDataTests/API/PythonModel/PythonModelTests.cs:136]: You can mock the url shortener service using the technique in [this test](https://github.com/bvcms/bvcms/blob/develop/UnitTests/eSpaceClientTests/eSpaceClientTests.cs#L16) for mocking the eSpace web service.

**Comment by davidcarroll** [comment]
@brondavies I tried to figure out how to mock the tpsdb.co/Create service but I wasted a lot of time trying to figure it out. I gave up and added a different Test for something else.

File: CmsData/API/PythonModel/PythonModel.Sms.cs [hunk 2]
```diff
@@ -24,5 +26,14 @@ public void SendSms(object query, int iSendGroup, string sTitle, string sMessage
             }
             TwilioHelper.QueueSms(db, query, iSendGroup, sTitle, sMessage);
         }
+        public static string CreateTinyUrl(string url)
+        {
+            var createTinyUrl = "https://tpsdb.co/Create";
```
Comment (brondavies) [inline CmsData/API/PythonModel/PythonModel.Sms.cs:31]: Move this to an AppSetting called "UrlShortenerService" and if it's not set, return the original `url`.

Comment (davidcarroll) [reply CmsData/API/PythonModel/PythonModel.Sms.cs]: Done

File: UnitTests/CmsDataTests/API/PythonModel/PythonModelTests.cs (hunk 1 above)
Comment (davidcarroll) [reply UnitTests/CmsDataTests/API/PythonModel/PythonModelTests.cs]: I gave up and added a unrelated unitest

File: CmsData/API/PythonModel/PythonModel.Sms.cs [hunk 3]
```diff
@@ -24,5 +26,14 @@ public void SendSms(object query, int iSendGroup, string sTitle, string sMessage
             }
             TwilioHelper.QueueSms(db, query, iSendGroup, sTitle, sMessage);
         }
+        public static string CreateTinyUrl(string url)
+        {
+            var createTinyUrl = ConfigurationManager.AppSettings["UrlShortenerService"] ?? "https://tpsdb.co/Create";
```
Comment (brondavies) [inline CmsData/API/PythonModel/PythonModel.Sms.cs:31]: @davidcarroll  We don't want to default to our service.  If the app setting is not there, just return the original url.

File: CmsWeb/Web.config [hunk 4]
```diff
@@ -62,6 +62,7 @@
     <add key="PushpayRU" value="touchpoint" />
     <!-- Pushpay Payment (use sandbox for test)-->
     <add key="PushpayGivingLinkBase" value="https://sandbox.pushpay.io/pay/" />
+    <add key="UrlShortenerService" value="https://tpsdb.co/Create"/>
```
Comment (brondavies) [inline CmsWeb/Web.config:65]: Leave this out of the base Web.config

**Comment by davidcarroll** [comment]
Where do I put it? I don't have access to the release configs

File: CmsData/API/PythonModel/PythonModel.Sms.cs (hunk 3 above)
Comment (davidcarroll) [reply CmsData/API/PythonModel/PythonModel.Sms.cs]: Done

File: CmsWeb/Web.config (hunk 4 above)
Comment (davidcarroll) [reply CmsWeb/Web.config]: Done

File: CmsData/API/PythonModel/PythonModel.Sms.cs [hunk 5]
```diff
@@ -24,5 +26,24 @@ public void SendSms(object query, int iSendGroup, string sTitle, string sMessage
...
+                var request = new RestRequest(Method.POST);
+                request.AddParameter("token", token);
+                request.AddParameter("url", url);
+                shorturl = client.Execute(request).Content;
+                if (string.IsNullOrEmpty(shorturl))
+                    return url;
+            }
+            if (shorturl.StartsWith("http"))
+                return shorturl;
+            return $"https://{shorturl}";
```
Comment (brondavies) [inline CmsData/API/PythonModel/PythonModel.Sms.cs:46]: You can save 8 characters by not including the protocol - almost all text messaging apps can still identify the url and create a clickable link.

Comment (davidcarroll) [reply CmsData/API/PythonModel/PythonModel.Sms.cs:46]: I was testing it on a webpage and the only way to get it to recognize it as a link was to include the protocol. I'll edit the code to remove it.

**Pull request merged**